test/com/google/enterprise/adaptor/filenet/PermissionsTest.java

```diff
@@ -691,4 +694,275 @@ public void testMarking_UserNotMatchingAnyAces() {
         SecurityPrincipalType.USER, "user2@bar.example.com", user1, false,
         AccessRight.NONE);
   }
+
+  //======================================================================
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_emptyAcl_allowConstraintMask() throws Exception {
+    Permissions testPerms = new Permissions(perms);
+    Permissions.Acl acl = testPerms.getMarkingAcl(~VIEW_ACCESS_RIGHTS);
+
+    assertEquals(ImmutableSet.<String>of(AUTHENTICATED_USERS),
+        acl.getAllowGroups(PermissionSource.MARKING));
+    assertEquals(ImmutableSet.<String>of(AUTHENTICATED_USERS),
+        acl.getAllowGroups());
+    assertTrue(acl.getAllowUsers().toString(), acl.getAllowUsers().isEmpty());
```

Comment:
More assertEquals? You're using it a bunch, just not in these empty assertions. Not really important, with the non-empty set printed, but a minor consistency point.